### PR TITLE
[HEL-6307] | Exposed the HeliumPresentationStyle as a config property

### PR DESCRIPTION
**Pull request by koyinusa**
> [!NOTE]
> **Medium Risk**
> Changes modal presentation and custom UIKit transitions where dismissal bugs could wedge the paywall presenter; mitigated by new unit tests but still user-visible presentation behavior.
> 
> **Overview**
> Adds a public **`HeliumPresentationStyle`** (`slideUp`, `slideLeft`) and an optional **`presentationStyle`** on **`PaywallPresentationConfig`** so apps can override the Helium dashboard animation for **`presentPaywall`** (not embedded / `.heliumPaywall`).
> 
> Presentation is applied via **`applyPresentationStyle`**: integrator choice wins; otherwise dashboard **`HeliumPaywallInfo.presentationStyle`** applies. **`slideLeft`** uses a custom slide-in transition; **`slideUp`** stays the default full-screen modal. **Cross-dissolve** and **flip** styles from the old internal enum are **removed**.
> 
> **`presentationStyle`** JSON decoding is **fault-tolerant** (unknown or malformed values decode as `nil`) so a bad animation string does not fail the whole paywall config.
> 
> The slide-left implementation is reworked (**`SlideInTransition`**) with a presentation controller that removes the presenter’s view (full-screen lifecycle), shadow performance tweaks, and safer dismiss / cancellation handling.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 5a976d641d1fc634958fbe2ecdf403369d5f606a. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

## Summary by CodeRabbit

- **New Features**
  - Added configurable paywall presentation styles via `PaywallPresentationConfig.presentationStyle` (slideUp, slideLeft).
  - Implemented a refined slide-left transition with correct UIKit transitioning behavior.
  - Integrator-selected style takes precedence over dashboard configuration.
- **Bug Fixes**
  - Improved decoding tolerance for presentation-style values so invalid/unknown inputs no longer break broader paywall configuration loading.
- **Tests**
  - Expanded coverage for presentation-style precedence, decoding edge cases, and slide transition presentation/dismissal/cancellation behavior.

### CODE REVIEW

File: Sources/Helium/HeliumPublic.swift [hunk 1]
```diff
@@ -452,29 +452,51 @@ public class Helium {
...
+    case slideUp
+
+    /// Slides in from the trailing edge on entry and back out on exit, matching a navigation push.
+    case slideLeft
+
+    /// Fades in over the presenting screen. Has no Helium Android counterpart.
+    case crossDissolve
+
+    /// Flips the presenting screen over horizontally. Has no Helium Android counterpart.
+    case flipHorizontal
```
Comment (salami) [inline Sources/Helium/HeliumPublic.swift]: The flip and cross dissolve are kinda weird and not sure if anyone will use, let's keep those unexposed for now or even remove

File: Sources/Helium/HeliumPaywallPresentation/SlideInTransition.swift [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+}
+
+/// Gives this style the same view lifecycle as the `.fullScreen` presentation every other style uses.
+/// A `.custom` presentation otherwise leaves the presenting view controller's view on screen, so that
+/// view controller never receives `viewWillDisappear`/`viewDidDisappear` while the paywall is up.
+private class SlideInPresentationController: UIPresentationController {
+    override var shouldRemovePresentersView: Bool { true }
+}
+
+/// The drop shadow that gives the sliding view its sense of depth over the content behind it.
```
Comment (salami) [inline Sources/Helium/HeliumPaywallPresentation/SlideInTransition.swift:37]: Does normal UINavigationController have a shadow? I don't know -- but I think we should be trying to look like that animation

**Review by salami — APPROVED** [review]

File: Sources/Helium/HeliumPublic.swift (hunk 1 above)
Comment (koyinusa) [reply Sources/Helium/HeliumPublic.swift]: I think we should just get rid of it.

File: Sources/Helium/HeliumPaywallPresentation/SlideInTransition.swift (hunk 2 above)
Comment (koyinusa) [reply Sources/Helium/HeliumPaywallPresentation/SlideInTransition.swift:37]: Yes it does. We can optimize the animation a bit. But that will a bigger change. 
Most of what we have here was me moving around the existing logic in the `PaywallPresentationStyle`
